app/src/components/features/rules/RuleBuilder/index.js

```diff
@@ -270,6 +274,19 @@ const RuleBuilder = (props) => {
     trackRuleEditorViewed(source, ruleType);
   }, [currentlySelectedRuleConfig.TYPE, state]);
 
+  useEffect(() => {
+    if (
+      MODE === RULE_EDITOR_CONFIG.MODES.EDIT &&
+      isDesktopOnlyRule(currentlySelectedRuleData)
+    ) {
+      if (
+        currentlySelectedRuleConfig.TYPE ===
+        GLOBAL_CONSTANTS.RULE_TYPES.REDIRECT
+      )
+        trackDesktopRuleViewedOnExtension("map_local");
+    }
+  }, [MODE, currentlySelectedRuleConfig.TYPE, currentlySelectedRuleData]);
+
```

Comment:
This does not ensure that desktop rule is viewed on extension. The second `if (
        currentlySelectedRuleConfig.TYPE ===
        GLOBAL_CONSTANTS.RULE_TYPES.REDIRECT
      )` is already checked in `isDesktopOnlyRule()`

The checks should be:
- Edit mode
- `isDesktopOnlyRule()`
- `appMode`